src/main/java/seedu/address/logic/parser/ViewCompanyCommandParser.java

```diff
@@ -17,12 +18,20 @@ public class ViewCompanyCommandParser implements Parser<ViewCompanyCommand> {
      */
     public ViewCompanyCommand parse(String args) throws ParseException {
 
-        String[] splitArgs = ParserUtil.parseRequiredNumberOfArguments(args, 2, ViewCompanyCommand.MESSAGE_USAGE);
-
+        String[] splitArgs = ParserUtil.parseRequiredNumberOfArguments(args,
+                2, ViewCompanyCommand.MESSAGE_USAGE);
         String indexString = splitArgs[1];
-        Index index = ParserUtil.parseIndex(indexString);
+        String entityString = splitArgs[0];
 
-        return new ViewCompanyCommand(index);
+        Index index = ParserUtil.parseIndex(indexString);
+        String entity = ParserUtil.parseEntity(entityString);
```

Comment:
This is very nitpicky, if we read from left to right, maybe we should check entity before index?

If not for that, we should do it for consistency sake since the other command parser follow this order.